crates/levm/tests/tests.rs

```diff
@@ -21,9 +23,394 @@ fn test() {
+    ];
+
+    let bytecode = operations
+        .iter()
+        .flat_map(Operation::to_bytecode)
+        .collect::<Bytes>();
+
+    vm.execute(bytecode);
+
+    assert!(vm.stack.pop().unwrap() == U256::from(2));
+}
+
+#[test]
+fn sdiv_op_for_zero() {
+    let mut vm = VM::default();
+
+    let operations = [
+        Operation::Push32(U256::zero()),
+        Operation::Push32(U256::one()),
+        Operation::Sdiv,
+        Operation::Stop,
+    ];
+
+    let bytecode = operations
+        .iter()
+        .flat_map(Operation::to_bytecode)
+        .collect::<Bytes>();
+
+    vm.execute(bytecode);
+
+    assert!(vm.stack.pop().unwrap() == U256::zero());
+}
+
+#[test]
+fn mod_op() {
+    let mut vm = VM::default();
+
+    let operations = [
+        Operation::Push32(U256::from(4)),
+        Operation::Push32(U256::from(10)),
+        Operation::Mod,
+        Operation::Stop,
+    ];
+
+    let bytecode = operations
+        .iter()
+        .flat_map(Operation::to_bytecode)
+        .collect::<Bytes>();
+
+    vm.execute(bytecode);
+
+    assert!(vm.stack.pop().unwrap() == U256::from(2));
+}
+
+#[test]
+fn mod_op_for_zero() {
+    let mut vm = VM::default();
+
+    let operations = [
+        Operation::Push32(U256::zero()),
+        Operation::Push32(U256::one()),
+        Operation::Mod,
+        Operation::Stop,
+    ];
+
+    let bytecode = operations
+        .iter()
+        .flat_map(Operation::to_bytecode)
+        .collect::<Bytes>();
+
+    vm.execute(bytecode);
+
+    assert!(vm.stack.pop().unwrap() == U256::zero());
+}
+
+#[test]
+fn smod_op() {
+    let mut vm = VM::default();
+
+    let operations = [
+        Operation::Push32(U256::from(0x03)),
+        Operation::Push32(U256::from(0x0a)),
+        Operation::SMod,
+        Operation::Stop,
+    ];
+
+    let bytecode = operations
+        .iter()
+        .flat_map(Operation::to_bytecode)
+        .collect::<Bytes>();
+
+    vm.execute(bytecode);
+
+    assert!(vm.stack.pop().unwrap() == U256::one());
+}
+
+#[test]
+fn smod_op_for_zero() {
+    let mut vm = VM::default();
+
+    let operations = [
+        Operation::Push32(U256::zero()),
+        Operation::Push32(U256::one()),
+        Operation::SMod,
+        Operation::Stop,
+    ];
+
+    let bytecode = operations
+        .iter()
+        .flat_map(Operation::to_bytecode)
+        .collect::<Bytes>();
+
+    vm.execute(bytecode);
+
+    assert!(vm.stack.pop().unwrap() == U256::zero());
+}
+
+#[test]
+fn addmod_op() {
+    let mut vm = VM::default();
+
+    let operations = [
+        Operation::Push32(U256::from(8)),
+        Operation::Push32(U256::from(0x0a)),
+        Operation::Push32(U256::from(0x0a)),
+        Operation::Addmod,
+        Operation::Stop,
+    ];
+
+    let bytecode = operations
+        .iter()
+        .flat_map(Operation::to_bytecode)
+        .collect::<Bytes>();
+
+    vm.execute(bytecode);
+
+    assert!(vm.stack.pop().unwrap() == U256::from(4));
+}
+
+#[test]
+fn addmod_op_for_zero() {
```

Comment:
Can you check that `ADDMOD` works correctly when adding `2^256 - 2` and `2^256 - 3` modulo `2^256 - 1`?

Reply:
Checked it and the code had a bug returning a wrong value, just fixed it and added a test 🫡